cmd/root.go

```diff
@@ -261,7 +261,9 @@ func restClientPool(cmd *cobra.Command, envName *string) (dynamic.ClientPool, di
 }
 
 type envSpec struct {
-	env   *string
+	env  *string
+	env2 *string
```

Comment:
I'm not sure this is the best place for `env2`. `envSpec` was semantically supposed to specify the files of a single environment. With two environment names, I'm not sure what `envSpec` "means" anymore.

Reply:
Hmm, what do you propose here? This is the sort of place where I wish go had options, but instead we have pointers.

On another note though, I'm not sure if that's what `envSpec` semantically specified even prior to this change.  The `files` in envSpec doesn't correspond to the env files (i.e. the expanded `dev.jsonnet`), but instead the contents of `-f` flag (empty if not provided).

Reply:
Ok, so when I say that "`envSpec` was semantically supposed to specify the files of a single environment", what I mean is that it's supposed to semantically represent the files for commands that take either _a set of `-f` flags_, or an _env name_, or _both_.

Adding a second environment makes every command that uses this data structure more complicated, because it has to know that `env2` is not important.

So fundamentally I think the problem here is that `parseEnvCmd` does not do what you want, because `diff` is no longer an "env command". What `diff` really wants is: (1) a single environment, (2) two environments, (3) a single environment with a `-f`, or (4) two environments with a `-f` flag. So in `diff` I'd just parse that logic.

It isn't awesome, but no one else needs this to be a general part of the abstraction.

Reply:
Makes sense, I'll make some changes. Just to clarify -- `diff` is true for the first 3 cases that you specified, but what is the behavior of "(4) two environments with a `-f` flag?" I don't think that case should be supported because it appears to be a 3-way diff.

Reply:
Is it a three-way merge? A command like `ks diff local:foo local:bar -f components/whatever` should diff those environments, but _only_ for that file, is what I was thinking.

Reply:
Follow up question would then be -- is `ks diff foo -f guestbook.jsonnet` diffing the resources in `guestbook.jsonnet` between the equivalent of `local:foo` or `remote:foo`?